modules/system-view/src/main/java/org/apache/ignite/internal/systemview/SystemViewManagerImpl.java

```diff
@@ -136,10 +181,78 @@ public Map<String, String> nodeAttributes() {
         return nodeAttributes;
     }
 
+    @Override
+    public void onNodeJoined(LogicalNode joinedNode, LogicalTopologySnapshot newTopology) {
+        processNewTopology(newTopology);
+    }
+
+    @Override
+    public void onNodeLeft(LogicalNode leftNode, LogicalTopologySnapshot newTopology) {
+        processNewTopology(newTopology);
+    }
+
+    @Override
+    public void onTopologyLeap(LogicalTopologySnapshot newTopology) {
+        processNewTopology(newTopology);
+    }
+
     /**
      * Returns future which is completed when system views are registered in the catalog.
      */
     public CompletableFuture<Void> completeRegistration() {
         return viewsRegistrationFuture;
     }
+
+    private void processNewTopology(LogicalTopologySnapshot topology) {
+        Map<String, List<String>> owningNodesByViewName = new HashMap<>();
+
+        for (LogicalNode logicalNode : topology.nodes()) {
+            String systemViewsNames = logicalNode.systemAttributes().get(NODE_ATTRIBUTES_KEY);
+
+            if (systemViewsNames == null) {
+                continue;
+            }
+
+            Arrays.stream(systemViewsNames.split(NODE_ATTRIBUTES_LIST_SEPARATOR))
+                    .map(String::trim)
+                    .forEach(viewName ->
+                            owningNodesByViewName.computeIfAbsent(viewName, key -> new ArrayList<>()).add(logicalNode.name())
+                    );
+        }
+
+        for (String viewName : owningNodesByViewName.keySet()) {
+            owningNodesByViewName.compute(viewName, (key, value) -> {
+                assert value != null;
+
+                return List.copyOf(value);
+            });
+        }
+
+        this.owningNodesByViewName = Map.copyOf(owningNodesByViewName);
+    }
+
+    private static class EmptyPublisher<T> implements Publisher<T> {
```

Comment:
from my point of view it will be better to inline the publisher realisaton instead of create separate the new static class. The main reason - we can forget to remove it during amend code above to right publisher

Reply:
> we can forget to remove it during amend code above to right publisher

I believe EmptyPublisher will be useful in any case

Reply:
replaced EmptyPublisher with throwing exception